backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/PredicatesHandler.java

```diff
@@ -94,6 +94,66 @@
     return InternalFuture.failedStage(new InvalidArgumentException("Predicate cannot be handled"));
   }
 
+  private InternalFuture<QueryFilterContext> processTagsPredicate(
+      long index, KeyValueQuery predicate) {
+    try {
+      final var value = predicate.getValue();
+      var operator = predicate.getOperator();
+      //  We will manage this operator at bottom of method based on `NE` and `NOT_CONTAIN` using
+      // `IN` OR `NOT IN` query
+      if (operator.equals(OperatorEnum.Operator.NOT_CONTAIN)) {
+        operator = OperatorEnum.Operator.CONTAIN;
+      } else if (operator.equals(OperatorEnum.Operator.NE)) {
+        operator = OperatorEnum.Operator.EQ;
+      }
```

Comment:
Please don't. This kind of solution made my life hell trying to understand the original code as things that were supposed to be the same became different. Just spend the extra typing to account for these types in the checks instead of just overwriting the operator and making `operator` be different from `predicate.getOperator()`. It's just not worth the mental burden for saving a handful of characters.

Reply:
Removed and modify as per sync up.